src/utilities/dates.ts

```diff
@@ -0,0 +1,186 @@
+export enum Weekday {
+  Sunday = 0,
+  Monday = 1,
+  Tuesday = 2,
+  Wednesday = 3,
+  Thursday = 4,
+  Friday = 5,
+  Saturday = 6,
+}
+
+export enum Month {
+  January = 0,
+  February = 1,
+  March = 2,
+  April = 3,
+  May = 4,
+  June = 5,
+  July = 6,
+  August = 7,
+  September = 8,
+  October = 9,
+  November = 10,
+  December = 11,
+}
+
+export interface Range {
+  start: Date;
+  end: Date;
+}
+
+export type Year = number;
+
+export type Week = (Date | null)[];
+
+const WEEK_LENGTH = 7;
+
+export function getWeeksForMonth(
+  month: Month,
+  year: Year,
+  weekStartsOn: Weekday = Weekday.Sunday,
+): Week[] {
+  const firstOfMonth = new Date(year, month, 1);
+  const firstDayOfWeek = firstOfMonth.getDay();
+  const weeks: Week[] = [[]];
+
+  let currentWeek = weeks[0];
+  let currentDate = firstOfMonth;
+
+  const orderedWeekday = getOrderedWeekdays(weekStartsOn);
+  for (let i = 0; i < orderedWeekday.indexOf(firstDayOfWeek); i++) {
+    currentWeek.push(null);
```

Comment:
Does this render blank squares in the date picker?

Reply:
Yep. The original dates file from javascript-utilities had this function but for some reason we were [using a duplicated version](https://github.com/Shopify/polaris-react/pull/3108/files#diff-5ef27a053fac0e45adb0486dbca21da4L80), so I updated that callsite to use this version instead